### PR TITLE
Operand form

### PR DESCRIPTION
**Pull request by swati-kale**
Added Operand form support, it will create CR using user inputs into the form.To navigate the form, Create Instance-> Edit Form. Please check examples in the 
tenant
api
api-mgr

 attached screen shots.

### CODE REVIEW

**Comment by bmozaffa** [comment]
Hi @eguzki and @miguelsorianod , this PR from @swati-kale takes advantage of a new feature in OpenShift 4.2 Console that optionally renders a web form for (CR) data input instead of the yaml editor. Please review and advise if any changes are needed.

**Comment by eguzki** [comment]
seems that updated CSV in no longer a valid yaml.

```
cd /home/circleci/project/deploy/olm-catalog && operator-courier verify --ui_validate_io 3scale-operator-master/
ERROR:operatorcourier.identify:Courier requires valid input YAML files
Courier requires valid input YAML files
```

Please, fix it

**Comment by eguzki** [comment]
Cool feature, indeed!

**Comment by miguelsorianod** [comment]
Hi @swati-kale thank you for your PR.

Would this new manifest YAML with this new fields be backwards-compatible with OpenShift 4.1? This is, could the manifest still be used in OpenShift 4.1 without problems?

How did you add this fields? Was it manually or through some tool? We are wondering about how this would be maintained / what would we have to do when there are changes on the available fields.

Thank you.

**Comment by swati-kale** [comment]
> Hi @swati-kale thank you for your PR.
> 
> Would this new manifest YAML with this new fields be backwards-compatible with OpenShift 4.1? This is, could the manifest still be used in OpenShift 4.1 without problems?
> 
> How did you add this fields? Was it manually or through some tool? We are wondering about how this would be maintained / what would we have to do when there are changes on the available fields.
> 
> Thank you.



> Hi @swati-kale thank you for your PR.
> 
> Would this new manifest YAML with this new fields be backwards-compatible with OpenShift 4.1? This is, could the manifest still be used in OpenShift 4.1 without problems?
> 
> How did you add this fields? Was it manually or through some tool? We are wondering about how this would be maintained / what would we have to do when there are changes on the available fields.
> 
> Thank you.

Hi @miguelsorianod,

It is compatible with 4.1 as it did not give any errors while creating instances, though the Form creation is available for 4.2 onwards only.
I added specDescriptors for these fields manually. As per design, all fields should come from OpenAPI schema from CRD, if not then those should be defined undere specdescriptors in CSV. The feature is still under development and hopefully when implemented very  less or no fields will be needed from CSV and all will be available through CRD only. 

Thanks!

**Comment by swati-kale** [comment]
> seems that updated CSV is no longer a valid yaml.
> 
> ```
> cd /home/circleci/project/deploy/olm-catalog && operator-courier verify --ui_validate_io 3scale-operator-master/
> ERROR:operatorcourier.identify:Courier requires valid input YAML files
> Courier requires valid input YAML files
> ```
> 
> Please, fix it

Sure will look into it. Thanks!

File: deploy/olm-catalog/3scale-operator-master/3scale-operator-master.v0.0.0.clusterserviceversion.yaml [hunk 1]
```diff
@@ -14,82 +14,955 @@ metadata:
...
+            displayName: Image
+            path: system.database.postgresql.image
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:system'
+              - 'urn:alm:descriptor:com.tectonic.ui:text'
+          - description: AWS Bucket
+            displayName: AWS Bucket
+            path: system.fileStorage.amazonSimpleStorageService.awsBucket
+            x-descriptors:
+              - 'urn:alm:descriptor:com.tectonic.ui:fieldGroup:system'
```
Comment (miguelsorianod) [inline deploy/olm-catalog/3scale-operator-master/3scale-operator-master.v0.0.0.clusterserviceversion.yaml]: Are nested fieldGroups configurable? Could this be placed inside the "amazonSimpleSotrageService" group?
How it looks currently from one of the captures provided this field appears under the .spec.system section but it's really under the .spec.system.fileStorage.amazonSimpleStorageService section.

Same situation applies to several other fields that are nested on nested fields.

Comment (swati-kale) [reply deploy/olm-catalog/3scale-operator-master/3scale-operator-master.v0.0.0.clusterserviceversion.yaml]: Yes, "nested fieldgroups" are not yet implemented. Even the component to add and delete into object arrays is not implemented, so as of now I have just added one element for it.
e.g " integrationMethod.apicastHosted.mappingRulesSelector.matchExpressions[0].key" (line #323 and many others).

**Comment by eguzki** [comment]
Closing as this PR has been stale for too long

**Pull request closed without merging**